### PR TITLE
Add rank

### LINKED ISSUE #134: Feature: Add Rank
Add `rank(x)` which returns the number of set bits <= x

### PR DESCRIPTION
**Pull request by saik0**
Implement `rank(x)` which returns the number of set bits <= x

Fixes #134

### CODE REVIEW

File: src/bitmap/inherent.rs [hunk 1]
```diff
@@ -358,6 +358,38 @@ impl RoaringBitmap {
...
+    pub fn rank(&self, value: u32) -> u64 {
+        // if len becomes cached for RoaringBitmap: return len if len > value
+
+        let (key, index) = util::split(value);
+
+        match self.containers.binary_search_by_key(&key, |c| c.key) {
+            Ok(i) => {
+                // For optimal locality of reference:
+                //  * container[i] should be a cache hit after binary search, rank it first
+                //  * sum in reverse to avoid cache misses near i
```
Comment (Kerollmops) [inline src/bitmap/inherent.rs:386]: That's pretty smart, I never thought about that before!

**Review by Kerollmops — APPROVED** [review]
That's a pretty good PR, thank you again!
bors merge

**Comment by Kerollmops** [comment]
CI doesn't pass [due to a bug](https://github.com/RoaringBitmap/roaring-rs/runs/4816876843?check_suite_focus=true#step:6:302).

**Comment by Kerollmops** [comment]
Thank you!
Bors merge

**Comment by Kerollmops** [comment]
Clippy is [a little bit picky here](https://github.com/RoaringBitmap/roaring-rs/runs/4822132063?check_suite_focus=true#step:9:27). Could you fix that, please?

**Comment by saik0** [comment]
Fixed

**Comment by Kerollmops** [comment]
Thank you!
bors merge

**Pull request merged**